### PR TITLE
fix(ci): prevent pipelines from being in blocked state

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
As per [documentation](https://docs.gitlab.com/ee/ci/yaml/#allow_failure:~:text=false%20for%20jobs%20that%20use%20when%3A%20manual%20inside%20rules.), `allow_failure` defaults to `false` when used with `when: manual`. So added a bunch of `allow_failure: true` to unblock PR builds

### Motivation
CI performance

### Additional Notes
Before
image
after
image

### CODE REVIEW

File: .gitlab/benchmarks/macrobenchmarks.yml
Comment (chouetz) [inline .gitlab/benchmarks/macrobenchmarks.yml:1]: In this file I removed the `when: manual` from the job definition as it is included in the rule each job extends.

**Review by chouquette — APPROVED** [review]
LGTM for BARX file

**Review by L3n41c — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

**Review by ddyurchenko — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Pull request merged**